### PR TITLE
Test casper appbar

### PR DESCRIPTION
**Pull request by rodriguez-facundo**
extends Casper test for appbar

### CODE REVIEW

File: components/settings/NetPyNETabs.js [hunk 1]
```diff
@@ -63,9 +64,9 @@ export default class NetPyNETabs extends React.Component {
                 targetOrigin={{ horizontal: "right", vertical: "top" }}
                 anchorOrigin={{ horizontal: "right", vertical: "bottom" }}
             >
-                <MenuItem primaryText="Create Network" value="Create Network" />
-                <MenuItem primaryText="Create and Simulate Network" value="Create and Simulate Network" />
-                <MenuItem primaryText="Explore Existing Network" value="Explore Existing Network" />
+                <MenuItem id="transit_not" primaryText="Create Network" value="Create Network" />
```
Comment (adrianq) [inline components/settings/NetPyNETabs.js]: should be transit_net?

Comment (adrianq) [reply components/settings/NetPyNETabs.js]: btw, what is the rule regarding id naming? camelcase or underscore?

File: components/settings/NetPyNEToolBar.js [hunk 2]
```diff
@@ -105,9 +74,9 @@ export default class NetPyNEToolBar extends React.Component {
             
         }
         
-      
         return <div>
             <IconButton
+								id="appbar"
```
Comment (adrianq) [inline components/settings/NetPyNEToolBar.js]: appbar or appBar or app_bar?

File: components/settings/NetPyNEToolBar.js [hunk 3]
```diff
@@ -52,48 +49,20 @@ export default class NetPyNEToolBar extends React.Component {
                         changeTab={this.props.changeTab}
                         />
                     break;
-                case 'ImportHLS':
+                case 'Import':
```
Comment (adrianq) [inline components/settings/NetPyNEToolBar.js]: I would still use 'ImportHLS' even if this is the only Import so we don't have to change it back in the future

File: components/settings/actions/ActionDialog.js [hunk 4]
```diff
@@ -31,7 +31,14 @@ export default class ActionDialog extends React.Component {
     performAction = () => {
         if (this.props.isFormValid === undefined || this.props.isFormValid()){
             GEPPETTO.trigger(GEPPETTO.Events.Show_spinner, this.props.message);
-            this.closeDialog();
+						this.closeDialog();
+						if (this.props.command == "netpyne_geppetto.deleteModel") {
```
Comment (adrianq) [inline components/settings/actions/ActionDialog.js]: why this? what is the purpose of this bit of code?

File: components/settings/actions/ActionDialog.js [hunk 5]
```diff
@@ -77,7 +84,7 @@ export default class ActionDialog extends React.Component {
                 var title = this.props.title
                 var actions = [
                     cancelAction, 
-                    <RaisedButton primary label={this.props.buttonLabel} onTouchTap={this.performAction}/>
+                    <RaisedButton id="appbarPerformActionButton" primary label={this.props.buttonLabel} onClick={this.performAction}/>
```
Comment (adrianq) [inline components/settings/actions/ActionDialog.js]: appbar now instead of appBar?

File: components/settings/actions/ImportExportHLS.js [hunk 6]
```diff
@@ -12,6 +12,7 @@ export default class ImportExportHLS extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
+						fileName: "output",
```
Comment (adrianq) [inline components/settings/actions/ImportExportHLS.js]: indentantion

File: tests/netpyne-tests.js [hunk 7]
```diff
@@ -21,7 +23,11 @@ casper.test.begin('NetPyNE projects tests', function suite(test) {
   casper.options.waitTimeout = 10000
   casper.on("page.error", function(msg, trace) {
     this.echo("Error: " + msg, "ERROR");
-  });
+	});
+	
+	// casper.on('remote.message', function(message) { // UNCOMMENT TO GET MORE DEBUGGING INFO THAN WHAT YOU PROBABLY WANT
```
Comment (adrianq) [inline tests/netpyne-tests.js]: what is this? what extra info do we get here?

File: tests/netpyne-tests.js [hunk 8]
```diff
@@ -152,8 +163,76 @@ function loadConsole(test, consoleButton, consoleContainer) {
     this.wait(1000)
   })
 }
-/*******************************************************************************
- *                                 popParams                                   *
+/******************************************************************************
+ *                                  appbar                                    *
+ ******************************************************************************/
+function testAppbar(test) {
```
Comment (adrianq) [inline tests/netpyne-tests.js]: looks like the indentation is a bit of a mess here

File: NetPyNE.js [hunk 9]
```diff
@@ -134,10 +134,13 @@ export default class NetPyNE extends React.Component {
 			return {
 				value: tab,
 				prevValue: tab, 
-				freezeInstance: args.freezeInstance!=undefined?args.freezeInstance:fi,
-				freezeSimulation: args.freezeSimulation!=undefined?args.freezeSimulation:fs,
+				freezeInstance: args.freezeInstance != undefined ? args.freezeInstance : fi,
+				freezeSimulation: args.freezeSimulation != undefined ? args.freezeSimulation : fs,
 			}
 		});
+		if (tab=="define") {
```
Comment (adrianq) [inline NetPyNE.js]: Is this new? what is this for? can we move it to another PR if it is not strictly required?

File: components/settings/actions/ActionDialog.js [hunk 10]
```diff
@@ -47,7 +54,7 @@ export default class ActionDialog extends React.Component {
                         GEPPETTO.trigger(GEPPETTO.Events.Hide_spinner);
                         this.props.onRequestClose();
                     }
-            });
+						});
```
Comment (adrianq) [inline components/settings/actions/ActionDialog.js]: btw, shouldn't we remove any use of 'NeuroML'? There is at least an 'if' clause using it in this file

**Review by adrianq — CHANGES_REQUESTED** [review]
@rodriguez-facundo It looks good to me. It is gonna be a huge improvement. I have dropped a few comments/questions. Can you please have a look at them?

File: components/settings/NetPyNETabs.js (hunk 1 above)
Comment (rodriguez-facundo) [reply components/settings/NetPyNETabs.js]: it would be exp (explore), cre (create), sim(simulate) .  which are the primaryText in the appbar dropdown menus

File: components/settings/actions/ActionDialog.js (hunk 4 above)
Comment (rodriguez-facundo) [reply components/settings/actions/ActionDialog.js]: if you plot in the simulation tab, and then you  click **delete model** in the appbar, all the plots appear on top of the definition tab.  They don't get destroyed. I used to handle passing props to destroy the plots, but after the refactor, we are no longer sending props to the action, so I came up with this to overcome the issue

File: tests/netpyne-tests.js (hunk 7 above)
Comment (rodriguez-facundo) [reply tests/netpyne-tests.js]: you capture the console.log() there. so if the console gives you an error because netpyne tried to import something and failed in the backend, you will see the error.
But this is super verbose, because it also captures other stuff. but it is useful because you can run casper instead of having to try yourself to understand what is failing

File: tests/netpyne-tests.js (hunk 8 above)
Comment (rodriguez-facundo) [reply tests/netpyne-tests.js]: I am not using eslint here, so I dont get corrections. maybe we can tackle the issue in a separate PR .

File: NetPyNE.js (hunk 9 above)
Comment (rodriguez-facundo) [reply NetPyNE.js]: that is to hide the widgets. we don't want all the plots in the define tab

File: components/settings/NetPyNETabs.js (hunk 1 above)
Comment (adrianq) [reply components/settings/NetPyNETabs.js]: so... are you changing it? Try to use the same rule for all the ids please: either camelcase or underscore. In general, it looks like you are using camelcase...

File: components/settings/actions/ActionDialog.js (hunk 4 above)
Comment (adrianq) [reply components/settings/actions/ActionDialog.js]: I have moved these lines inside the 'then' clause along with the other similar actions. I am also deleting widget in a more elegant way now. https://github.com/MetaCell/geppetto-netpyne/pull/81/commits/18880bb404cec5d9393be5eafd4a89c141d55765

**Review by adrianq — APPROVED** [review]
